### PR TITLE
Fixed nested curly braces parsing

### PR DESCRIPTION
**Pull request by sergey-nbl**
Fix for https://github.com/electricimp/Builder/issues/13

### CODE REVIEW

File: src/AstParser.js [hunk 1]
```diff
@@ -207,7 +207,7 @@ class AstParser {
     let matches;
 
     // extract source fragments and inline expressions
-    while (matches = /@{(.*?)}/.exec(fragment)) {
+    while (matches = /@{([^@\r\n]*)}/.exec(fragment)) {
```
Comment (ppetrosh) [inline src/AstParser.js]: Expression `@{set('')}{()}` will be parsed incorrectly.

**Comment by ppetrosh** [comment]
When it's done, please make sure to provide some tests covering this case. Thanks!

File: spec/AstParser/basic.spec.js [hunk 2]
```diff
@@ -197,4 +197,55 @@ describe('AstParser', () => {
       }
     ]);
   });
+
+  it('should parse enclosed curly brackets', () => {
+    const res = p.parse
+(`@macro set(val)
```
Comment (ppetrosh) [inline spec/AstParser/basic.spec.js:203]: Why don't you want to add it to the fixtures?

File: src/AstParser.js (hunk 1 above)
Comment (ppetrosh) [reply src/AstParser.js]: Another problem with this change is that it prevents the code fragment from containing the `@` symbol.

**Comment by sergey-nbl** [comment]
Please confirm, what output should be here?
  3 @macro set(val)
  4 local var = @{val}
  5 @end
  6
  7 @{set("{};");{local b = http};}} 

I've got 
local var = {};local b = http};}}

**Pull request merged**